<?xml version="1.0"?>
<launch>
  <!-- arguments -->
  <param name="/use_sim_time" value="true" />
  <arg name="nodelet_manager" default="velodyne_nodelet_manager" />
  <!-- in case you use velodyne_driver, comment out the following line -->
  <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)" args="manager" output="screen"/>

    <!-- globalmap_server_nodelet -->
    <node pkg="nodelet" type="nodelet" name="globalmap_server_nodelet" args="load localization/GlobalmapServerNodelet $(arg nodelet_manager)">
     <!--hall.pcd lab_1201.pcd-->  
     <param name="globalmap_pcd" value="$(find localization)/data/autoware_gym1.pcd" />
     <param name="downsample_resolution" value="0.4" />
     <!--if use loam, need to transform axis. autoware, loam -->
     <param name="slam_method" value="autoware" />
    </node>

    <!-- localization_nodelet -->
    <node pkg="nodelet" type="nodelet" name="localization_nodelet" args="load localization/LocalizationNodelet $(arg nodelet_manager)">
        <!-- imu settings -->
        <!-- during "cool_time", imu inputs are ignored -->
        <param name="use_imu" value="false" />
        <param name="invert_imu" value="false" />
        <param name="cool_time_duration" value="2.0" />
        <!-- ndt settings -->
        <!-- if NDT is slow for your PC, try DIRECT1 serach method, which is a bit unstable but extremely fast -->
        <param name="ndt_neighbor_search_method" value="DIRECT7" />Q	
	<!--voxel的边长大小,过小造成内存占用过高,过大会导致误差太大-->
	<param name="ndt_resolution" value="3" />   
        <param name="downsample_resolution" value="0.3" />
        <!-- if "specify_init_pose" is true, pose estimator will be initialized with the following params -->
        <!-- otherwise, you need to input an initial pose with "2D Pose Estimate" on rviz" -->
        <param name="specify_init_pose" value="true" />
        <param name="init_pos_x" value="0.0" />
        <param name="init_pos_y" value="0.0" />
        <param name="init_pos_z" value="0.0" />
        <param name="init_ori_w" value="1.0" />
        <param name="init_ori_x" value="0.0" />
        <param name="init_ori_y" value="0.0" />
        <param name="init_ori_z" value="0.0" />
    </node>
     <node pkg="rviz" type="rviz" name="rviz" args="-d $(find localization)/rviz/localization.rviz" />
</launch>
